### PR TITLE
imp: Add STANDARD_HEADERS list

### PR DESCRIPTION
**Pull request by sdelamo**
STANDARD_HEADERS contains every HttpHeader constant

see: https://github.com/micronaut-projects/micronaut-azure/pull/437

### CODE REVIEW

File: http/src/main/java/io/micronaut/http/HttpHeaders.java [hunk 1]
```diff
@@ -499,6 +499,103 @@ public interface HttpHeaders extends Headers {
      */
     String X_AUTH_TOKEN = "X-Auth-Token";
 
+    List<String> STANDARD_HEADERS = Arrays.asList(
```
Comment (graemerocher) [inline http/src/main/java/io/micronaut/http/HttpHeaders.java]: should be immutable

**Review by graemerocher — CHANGES_REQUESTED** [review]

File: http/src/main/java/io/micronaut/http/HttpHeaders.java (hunk 1 above)
Comment (graemerocher) [reply http/src/main/java/io/micronaut/http/HttpHeaders.java]: and should have javadoc

Comment (sdelamo) [reply http/src/main/java/io/micronaut/http/HttpHeaders.java]: https://github.com/micronaut-projects/micronaut-core/pull/8903/commits/998f634d13f9d15a91e9cab4d5e233cfc1525b27 and https://github.com/micronaut-projects/micronaut-core/pull/8903/commits/ec236c48dc22339a9b60a62930b01ffa534de15d

File: http/src/main/java/io/micronaut/http/HttpHeaders.java [hunk 2]
```diff
@@ -499,6 +499,106 @@ public interface HttpHeaders extends Headers {
      */
     String X_AUTH_TOKEN = "X-Auth-Token";
 
+    /**
+     * Unmodifiable List of every header define in {@link HttpHeaders}.
```
Comment (graemerocher) [inline http/src/main/java/io/micronaut/http/HttpHeaders.java]:
```suggestion
     * Unmodifiable List of every header constant defined in {@link HttpHeaders}.
```

**Review by graemerocher — APPROVED** [review]

**Pull request merged**